src/ess/reduce/nexus.py

```diff
@@ -61,6 +70,7 @@ def load_detector(
     *,
     detector_name: NeXusDetectorName,
     entry_name: Optional[NeXusEntryName] = None,
+    definitions: Optional[Mapping] = None,
```

Comment:
Please check, but I think this is *not* the correct default (I hope the unit tests catch this?). Setting the definitions to `None` will override the default definitions, see https://github.com/scipp/scippnexus/blob/1cfab2a69a541832f83d088355eb6fa413a896c8/src/scippnexus/file.py#L15.

Reply:
Did you check how the argument is used?
See https://github.com/scipp/essreduce/pull/36/files#diff-a0ef1d672f81176717b7e31a2df573d29b3087dd28ce7eb354e1570c78b811b5R298-R300.

Does that still override the default?

Reply:
... but then the user cannot set "no definitions"? That may be a valid use case.

Reply:
Wouldn't that be done by `definitions={}`? Or do you mean that `definitions=None` has a specific meaning in scippnexus?

Reply:
In ScippNexus, using `definitions=None` and `definitions={}` is equivalent. It would be confusing if ESSreduce behaved differently (using `definitions=None` to mean "default definitions").

Reply:
Okay, is there a reason why `scippnexus` doesn't follow the convention that `None` means "use the default value".

Is it correct then that the default value for the `definitions` arguments in essreduce needs to be `snx.base_definitions()`?

Reply:
Okay, however that causes an issue in this PR. Because if we set the default value of `definitions` to something that is not `None` then we cannot distinguish the case when no definitions were provided from the case when some definitions were provided.

That becomes in issue if the `file_path` is an already opened nexus file (that is, if `file_path` is an `snx.Group`), because if new definitions are provided when we are loading from the group, we have to raise, but if no new definitions were provided we can proceed.

Reply:
@jl-wynen made the nice suggestion to create a new default value type and use that, seems to solve the problem :+1: